modules/next/src/Plugin/Next/PreviewUrlGenerator/SimpleOauth.php

```diff
@@ -159,66 +159,19 @@ public function validate(Request $request) {
       throw new InvalidPreviewUrlRequest("The provided secret has expired.");
     }
 
-    if (empty($body['scope'])) {
-      throw new InvalidPreviewUrlRequest("Field 'scope' is missing");
-    }
-
     // Validate the secret.
     if (empty($body['secret'])) {
       throw new InvalidPreviewUrlRequest("Field 'secret' is missing");
     }
 
-    if ($body['secret'] !== $this->previewSecretGenerator->generate($body['timestamp'] . $body['slug'] . $body['scope'] . $body['resourceVersion'])) {
+    if ($body['secret'] !== $this->previewSecretGenerator->generate($body['timestamp'] . $body['slug'] . $body['resourceVersion'])) {
       throw new InvalidPreviewUrlRequest("The provided secret is invalid.");
     }
 
     return [
-      'scope' => $body['scope'],
+      'path' => $body['slug'],
+      'maxAge' => (int) $this->configuration['secret_expiration'],
```

Comment:
For a project I pass along the maxAge and path to be able to set it on the setPreviewdata (see: https://nextjs.org/docs/pages/building-your-application/configuring/preview-mode#specify-the-preview-mode-duration). This way Drupal can have control over the duration of the preview session and the preview-mode is only enabled on the specified path. This introduces a better UX, because you don't see the exit-preview bar on every page you visit, you only see it on the page you requested the preview. This implementation requires a small change to the the preview function in modules/next package.

```
const validationPayload = await result.json()
const { path, maxAge } = validationPayload

response.setPreviewData(
  {
    resourceVersion,
    plugin,
    ...validationPayload,
  },
  {
    path,
    maxAge,
  },
)
```

If this change is desirable I will add the above code, otherwise I will remove the magAge from this PR.

Reply:
The `path` and `maxAge` options of `setPreviewData()` get converted into the `path` and `expires` fields of the preview cookie. Next.js's `draftMode().enable()` does not have any options to set those fields on the draft cookie.

Also, `maxAge` from the Drupal Next module is about how long the validation secret can be used to validate the draft/preview request. Re-using `maxAge` to determine when the preview/draft cookie expires feels like a mis-match of usage to me.

But as I said before:
> To be fair, the only thing the client is going to use is the 200 OK response code, so it doesn't really matter that much.

So I'm fine with leaving `maxAge` in the response.